src/ape_run/_cli.py

```diff
@@ -1,33 +1,152 @@
+    }
+
+
+class ScriptCommand(click.MultiCommand):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs, result_callback=self.result_callback)
+
+    def __load_module(self, filepath: Path) -> Optional[ModuleType]:
+        try:
+            # TODO: Analyze security issues from this
+            return SourceFileLoader("script", str(filepath)).load_module()
+
+        except Exception as err:
+            relative_filepath = get_relative_path(filepath, self._project.path)
+            logger.error_from_exception(err, f"Exception while loading script: {relative_filepath}")
+            return None
+
+    def _get_command(self, filepath: Path) -> Union[click.Command, click.Group, None]:
+        relative_filepath = get_relative_path(filepath, self._project.path)
+
+        # First load the code module by compiling it
+        # NOTE: This does not execute the module
+        logger.debug(f"Parsing module: {relative_filepath}")
+        try:
+            code = compile(filepath.read_text(), filepath, "exec")
+        except SyntaxError as e:
+            logger.error_from_exception(e, f"Exception while parsing script: {relative_filepath}")
+            return None  # Prevents stalling scripts
+
+        # NOTE: Introspect code structure only for given patterns (do not execute it to find hooks)
+        if "cli" in code.co_names:
+            # If the module contains a click cli subcommand, process it and return the subcommand
+            logger.debug(f"Found 'cli' command in script: {relative_filepath}")
+
+            script = self.__load_module(filepath)
+            if script:
+                return script.cli  # type: ignore
+
+            return None  # NOTE: Allow other scripts to load if loading script breaks
+
+        elif "main" in code.co_names:
+            logger.debug(f"Found 'main' method in script: {relative_filepath}")
+
+            @click.command(short_help=f"Run '{relative_filepath}:main'")
+            def call():
+                module = self.__load_module(filepath)
+                if module:
+                    module.main()  # type: ignore
+
+            return call
+
+        else:
+            logger.warning(f"No 'main' method or 'cli' command in script: {relative_filepath}")
+
+            @click.command(short_help=f"Run '{relative_filepath}'")
+            def call():
+                self.__load_module(filepath)
+                # Nothing to call
+
+            return call
+
+    @cached_property
+    def _project(self) -> ProjectManager:
+        """
+        A class representing the project that is active at runtime.
+        (This is the same object as from ``from ape import project``).
 
+        Returns:
+            :class:`~ape.managers.project.ProjectManager`
+        """
 
-@click.command(cls=NetworkBoundCommand, short_help="Run scripts from the `scripts` folder")
-@click.argument("scripts", nargs=-1)
+        from ape import project
+
+        project.config.load()
```

Comment:
```suggestion
        project.config_manager.load()
```

(if this is still needed)